deployment/src/main/java/io/quarkiverse/cxf/deployment/QuarkusCxfProcessor.java

```diff
@@ -212,93 +222,94 @@ public void build(
         }
 
-        //TODO bad code it is set in loop but use outside...
-        ClassOutput classOutput = new GeneratedBeanGizmoAdaptor(generatedBeans);
-        quarkusCapture c = new quarkusCapture(classOutput);
+        Bus bus = BusFactory.getDefaultBus();
+        // setup class capturing
+        bus.setExtension(new quarkusCapture(new GeneratedBeanGizmoAdaptor(generatedBeans)),
+                GeneratedClassClassLoaderCapture.class);
+
+        Set<String> clientSEIsInUse = findClientSEIsInUse(index);
 
         for (AnnotationInstance annotation : index.getAnnotations(WEBSERVICE_ANNOTATION)) {
             if (annotation.target().kind() != AnnotationTarget.Kind.CLASS) {
                 continue;
             }
 
             ClassInfo wsClassInfo = annotation.target().asClass();
+
             String sei = wsClassInfo.name().toString();
+
             reflectiveClass
                     .produce(new ReflectiveClassBuildItem(true, true, sei));
             unremovableBeans.produce(new UnremovableBeanBuildItem(
                     new UnremovableBeanBuildItem.BeanClassNameExclusion(sei)));
-            List<String> wrapperClassNames = new ArrayList<>();
 
             if (!Modifier.isInterface(wsClassInfo.flags())) {
                 continue;
             }
-            QuarkusJaxWsServiceFactoryBean jaxwsFac = new QuarkusJaxWsServiceFactoryBean();
-            Bus bus = BusFactory.getDefaultBus();
-            jaxwsFac.setBus(bus);
-            bus.setExtension(c, GeneratedClassClassLoaderCapture.class);
-            //TODO here add all class
-            try {
-                jaxwsFac.setServiceClass(Thread.currentThread().getContextClassLoader().loadClass(sei));
-                jaxwsFac.create();
-                wrapperClassNames.addAll(jaxwsFac.getWrappersClassNames());
-            } catch (ClassNotFoundException e) {
-                LOGGER.error("failed to load WS class : " + sei);
-            }
-            String pkg = wsClassInfo.name().toString();
-            int idx = pkg.lastIndexOf('.');
-            if (idx != -1 && idx < pkg.length() - 1) {
-                pkg = pkg.substring(0, idx);
-            }
-            AnnotationValue namespaceVal = annotation.value("targetNamespace");
-            String wsNamespace = namespaceVal != null ? namespaceVal.asString() : getNamespaceFromPackage(pkg);
-            String wsName = "";
-            if (annotation.value("serviceName") != null) {
-                wsName = annotation.value("serviceName").asString();
-            }
-            Collection<ClassInfo> implementors = index.getAllKnownImplementors(DotName.createSimple(sei));
+
+            // created on-demand if an implementor or client usage is found
+            QuarkusJaxWsServiceFactoryBean factoryBean = null;
+
+            String wsNamespaceDefault = Optional.ofNullable(annotation.value("targetNamespace"))
```

Comment:
why call it default if it is always the one we use ? wsNamespace seems fine.

Reply:
I did this because I had a version where overwrite effects could happen by multiple implementors. But it doesn't seem to be a problem anymore.
I'll re-check.

Reply:
done